mm2src/mm2_test_helpers/dummy_files/iris_test_history.json

```diff
@@ -0,0 +1,170 @@
+[
+    {
+        "tx_hex": "0a2a696161317a36303967397a386566346a6c6b63666767333663326c6a67636d6e6135306c32796a7261301240384636304534433336384335344539343233344145424135394331344531384146413138443437303530314434383944444641364539363635444135323738451a4064363034613132626531616239643966343531306435343832666232323762613530363238373531326466333561626337303730313763366439306165303837",
+        "tx_hash": "6BA67D4C7A0F48E5D85EE18EF5A649ED456B539D902828D56849B57DCFF5914B",
+        "from": [
+            "iaa1z609g9z8ef4jlkcfgg36c2ljgcmna50l2yjra0"
+        ],
+        "to": [
+            "iaa1erfnkjsmalkwtvj44qnfr2drfzdt4n9ldh0kjv"
+        ],
+        "total_amount": "0.026427",
+        "spent_by_me": "0.026427",
+        "received_by_me": "0",
+        "my_balance_change": "-0.026427",
+        "block_height": 6148319,
+        "timestamp": 0,
+        "fee_details": {
+            "type": "Utxo",
+            "coin": "IRIS-TEST",
+            "amount": "0.026426"
+        },
+        "coin": "IRIS-TEST",
+        "internal_id": "4538314545353844354538344630413743344437364142360000000000000000",
+        "transaction_type": "StandardTransfer"
+    },
+    {
+        "tx_hex": "0a2a696161317a36303967397a386566346a6c6b63666767333663326c6a67636d6e6135306c32796a726130122a696161316572666e6b6a736d616c6b7774766a3434716e6672326472667a6474346e396c6468306b6a762a0a0a05756e79616e12013132403834373331663565393739643965616263343838393234356131613165663932396430386437383832353163353133633834303632373535313032343765326540e807",
+        "tx_hash": "F1C633D16C1578A16450555A06349989A4B1E2223D7AAA2FE12396A5F855FD9B",
+        "from": [
+            "iaa1z609g9z8ef4jlkcfgg36c2ljgcmna50l2yjra0"
+        ],
+        "to": [
+            "iaa1erfnkjsmalkwtvj44qnfr2drfzdt4n9ldh0kjv"
```

Comment:
In `create_htlc` case, the locked amount is in fact transferred to a magic address:
```json
{
  "type":"transfer",
  "attributes":[
    {
      "key":"recipient",
      "value":"iaa1a7xynj4ceft8kgdjr6kcq0s07y3ccya6mepzdm"
    },
    {
      "key":"sender",
      "value":"iaa1z609g9z8ef4jlkcfgg36c2ljgcmna50l2yjra0"
    },
    {
      "key":"amount",
      "value":"1unyan"
    }
  ]
}
```

`iaa1erfnkjsmalkwtvj44qnfr2drfzdt4n9ldh0kjv` will only receive them if HTLC is successfully claimed.

Reply:
Explorers show it in that way. Which approach should we apply in our implementation? Should I skip the htlcs if we are the receiver?

Reply:
Explorer shows `to` of `CreateHtlc` message https://nyancat.iobscan.io/#/txs/F1C633D16C1578A16450555A06349989A4B1E2223D7AAA2FE12396A5F855FD9B, this shouldn't be mixed up with `to` field of standard transfer like here https://nyancat.iobscan.io/#/txs/49C155B133A171B3E43C4008C5E22DB03939F4DB8FF502CFF0E9A996D219B1B4.
If someone else sends `CreateHtlc` setting our address as `to/receiver` - coins are locked on intermediate address, they do not reach ours, so this tx shouldn't be showed. Actual incoming transfer will happen only when our address successfully claims the HTLC.